devices/vesternet.js

```diff
@@ -0,0 +1,188 @@
+const exposes = require('../lib/exposes');
+const fz = require('../converters/fromZigbee');
+const tz = require('../converters/toZigbee');
+const reporting = require('../lib/reporting');
+const extend = require('../lib/extend');
+const e = exposes.presets;
+
+module.exports = [
+    {
+        zigbeeModel: ['HK-SL-DIM-A'],
+        model: 'VES-ZB-DIM-004',
+        vendor: 'Vesternet',
+        description: 'Zigbee dimmer',
+        fromZigbee: extend.light_onoff_brightness().fromZigbee
+            .concat([fz.electrical_measurement, fz.metering, fz.ignore_genOta, fz.power_on_behavior]),
+        toZigbee: extend.light_onoff_brightness().toZigbee.concat([tz.power_on_behavior]),
+        exposes: [e.light_brightness(), e.power(), e.voltage(), e.current(), e.energy(), e.power_on_behavior(['off', 'on', 'previous'])],
+        whiteLabel: [{vendor: 'Sunricher', model: 'SR-ZG9040A'}],
+        configure: async (device, coordinatorEndpoint, logger) => {
+            const endpoint = device.getEndpoint(1);
+            await reporting.bind(endpoint, coordinatorEndpoint, ['genOnOff', 'genLevelCtrl', 'haElectricalMeasurement', 'seMetering']);
+            await reporting.onOff(endpoint);
+            await reporting.brightness(endpoint);
+            await reporting.readEletricalMeasurementMultiplierDivisors(endpoint);
+            await reporting.activePower(endpoint);
+            await reporting.rmsCurrent(endpoint, {min: 10, change: 10});
+            await reporting.rmsVoltage(endpoint, {min: 10});
+            await reporting.readMeteringMultiplierDivisor(endpoint);
+            await reporting.currentSummDelivered(endpoint);
+        },
+    },
+    {
+        zigbeeModel: ['ON/OFF -M'],
+        model: 'VES-ZB-HLD-017',
+        vendor: 'Vesternet',
+        description: 'Zigbee high load switch',
+        fromZigbee: [fz.on_off, fz.electrical_measurement, fz.metering, fz.power_on_behavior, fz.ignore_genOta],
+        toZigbee: [tz.on_off, tz.power_on_behavior],
+        exposes: [e.switch(), e.power(), e.current(), e.voltage(), e.energy(), e.power_on_behavior(['off', 'on', 'previous'])],
+        whiteLabel: [{vendor: 'Sunricher', model: 'SR-ZG9101SAC-HP-SWITCH-B'}],
+        configure: async (device, coordinatorEndpoint, logger) => {
+            const endpoint = device.getEndpoint(1);
+            await reporting.bind(endpoint, coordinatorEndpoint, ['genOnOff', 'haElectricalMeasurement', 'seMetering']);
+            await reporting.onOff(endpoint);
+            await reporting.readEletricalMeasurementMultiplierDivisors(endpoint);
+            await reporting.activePower(endpoint);
+            await reporting.rmsCurrent(endpoint, {min: 10, change: 10});
+            await reporting.rmsVoltage(endpoint, {min: 10});
+            await reporting.readMeteringMultiplierDivisor(endpoint);
+            await reporting.currentSummDelivered(endpoint);
+        },
+    },
+    {
+        zigbeeModel: ['HK-ZCC-A'],
```

Comment:
This device is already supported: https://github.com/Koenkk/zigbee-herdsman-converters/blob/587a9b8362f29e1620209e6480d0e5ad4a5b85cd/devices/sunricher.js#L344

Reply:
How best to handle this when different vendors use same base hardware manufacturer? This converter seems to work perfectly with the Vesternet variant, but is slightly different to the existing converter as it has coverInverted: true.

Reply:
Can you provide the data/database.db entry of this device?

Reply:
Here you go:

`{"id":2,"type":"Router","ieeeAddr":"0x84ba20fffed7cd7d","nwkAddr":53306,"manufId":4644,"manufName":"Sunricher","powerSource":"Mains (single phase)","modelId":"HK-ZCC-A","epList":[1,242],"endpoints":{"1":{"profId":260,"epId":1,"devId":515,"inClusterList":[0,3,4,5,6,8,258,2821,4096],"outClusterList":[25],"clusters":{"genBasic":{"attributes":{"modelId":"HK-ZCC-A","manufacturerName":"Sunricher","powerSource":1,"zclVersion":3,"appVersion":0,"stackVersion":0,"hwVersion":1,"dateCode":"NULL","swBuildId":"2.5.3_r48"}},"closuresWindowCovering":{"attributes":{"currentPositionLiftPercentage":0}}},"binds":[{"cluster":258,"type":"endpoint","deviceIeeeAddress":"0x0c4314fffe599345","endpointID":1}],"configuredReportings":[{"cluster":258,"attrId":8,"minRepIntval":1,"maxRepIntval":62000,"repChange":1}],"meta":{}},"242":{"profId":41440,"epId":242,"devId":102,"inClusterList":[33],"outClusterList":[33],"clusters":{},"binds":[],"configuredReportings":[],"meta":{}}},"appVersion":0,"stackVersion":0,"hwVersion":1,"dateCode":"NULL","swBuildId":"2.5.3_r48","zclVersion":3,"interviewCompleted":true,"meta":{"configured":424359850},"lastSeen":1676626023913,"defaultSendRequestWhen":"immediate"}`

Reply:
The `manufacturername` is the same (`Sunricher`), so we can only vary on the `swBuildId`. Replace the `zigbeeModel` line with:

```js
fingerprint: [{modelID: 'HK-ZCC-A', softwareBuildID: '2.5.3_r48'}],
```

Reply:
Thanks for the hint :-)

I've made this change for all the devices (with their respective swBuildId)  so hopefully that will avoid any other potential clashes.